topo_processor/stac/lds_cache.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+import pystac
+
+from topo_processor.util.aws_files import build_s3_path, load_file_content, s3_download
+from topo_processor.util.configuration import lds_cache_local_tmp_folder
+from topo_processor.util.files import decompress_file, is_gzip_file
+
+
+class LdsCache:
+    bucket: str
+    role: str
+
+    def __init__(self, bucket_name: str, role_arn: str):
+        self.bucket = bucket_name
+        self.role = role_arn
+
+    def get_layer(self, layer: str, version_number: str = None) -> str:
```

Comment:
Do we want to return the metadata csv as a `buffer`? Currently returning the path to the local temporary metadata file (which is deleted at the end of the process). I've made the system downloading the file as it's supposed to be `gzip` so I could not load the file content as I do for the `collection.json`